lib/iris/fileformats/grib/message.py

```diff
@@ -114,9 +115,19 @@ def sections(self):
         return self._raw_message.sections
 
     @property
+    def bmdi(self):
+        # Not sure of any cases where GRIB provides a fill value.
+        # Default for fill value is None.
```

Comment:
Our handling of missingValue is very strange.

GRIB1 loading: we set the `missingValue` key in the file to be `np.nan` before we load the data, so when we get the data values from the file it loads in the data array with nans where the missing values would be
GRIB1 saving: not supported
GRIB2 loading: we never use the `missingValue` key on load. Perhaps this is the missing link?
GRIB2 saving: we set the `missingValue` key to be the fill value of the cube.data as the link you added shows. Although, now with the dask stuff it should be the fill_value on the cube rather.